src/EVTUI/Core/Rendering/SceneManager.cs

```diff
@@ -0,0 +1,258 @@
+    ////////////////////////////////
+    public static void CheckGLError()
+    {
+        int err;
+        while ((err = (int)GL.GetError()) != 0)
+        {
+            ReportError(err);
+        }
+    }
+
+    public static void ReportError(int errorcode)
+    {
+        switch(errorcode)
+        {
+            case 0x500:
+                Console.WriteLine("OpenGLError: Invalid Enum");
+                break;
+            default:
+                Console.WriteLine("OpenGLError Code: 0x" + String.Format("{0:X}", errorcode));
+                break;
+        }
+    }
+
+    ///////////////////////////////
+    // *** Cleanup Functions *** //
+    ///////////////////////////////
+    public void teardown()
+    {
+        for (int i=this.sceneModels.Count-1; i>=0; --i)
+            this.UnloadModel(i);
+        this.sceneModels.Clear();
+        for (int i=this.externalGAPs.Count-1; i>=0; --i)
+            this.UnloadGAP(i);
+        this.externalGAPs.Clear();
+        for (int i=this.shaders.Count-1; i>=0; --i)
+            this.UnloadShader(i);
+        this.shaders.Clear();
+    }
+
+    /////////////////////////////////////
+    // *** Model Memory Management *** //
+    /////////////////////////////////////
+    public void LoadModel(string filepath)
+    {
+        if (!File.Exists(filepath))
+            throw new FileNotFoundException($"Model file '{filepath}' does not exist.");
+        
+        var model = GFDLibrary.Api.FlatApi.LoadModel(filepath);
+        var glmodel = new GLModel(model, ( material, textureName ) =>
+        {
+            if ( model.Textures.TryGetTexture( textureName, out var texture ) )
+            {
+                return new GLTexture( texture );
+            }
+            else
+            {
+                Trace.TraceWarning( $"tTexture '{textureName}' used by material '{material.Name}' is missing" );
+                return new GLTexture(Texture.CreateDefaultTexture(textureName));
+            }
+        } );
+
+        // // This should work with field models too...
+        // // But need to find a value for mIsFieldModel and mFieldTextures first.
+        // var glmodel = new GLModel(model, ( material, textureName ) =>
+        // {
+        //     if ( mIsFieldModel && mFieldTextures.TryOpenFile( textureName, out var textureStream ) )
+        //     {
+        //         using ( textureStream )
+        //         {
+        //             var texture = new FieldTexturePS3( textureStream );
+        //             return new GLTexture( texture );
+        //         }
+        //     }
+        //     else if ( model.Textures.TryGetTexture( textureName, out var texture ) )
+        //     {
+        //         return new GLTexture( texture );
+        //     }
+        //     else
+        //     {
+        //         Trace.TraceWarning( $"tTexture '{textureName}' used by material '{material.Name}' is missing" );
+        //     }
+
+        //     return null;
+        // } );
```

Comment:
What's the status on this commented block? Do field models work without it?

Reply:
They will work, but will only display dummy textures until I implement a clean way to deal with the external texture archives, which I thought would be best left for a later PR so that we can get some basic rendering elements in sooner.

Reply:
Yeah, totally agree. Getting field textures to work feels right as a separate PR.

Reply:
Are you able to clean up this commented block so it reads more clearly as a TODO?

Reply:
Fixed (hopefully)

Reply:
Yeah, looks good!